File: client/python/mesh/prsim/registry.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#

from abc import ABC, abstractmethod
from typing import List

from mesh.kinds import Registration
from mesh.macro import mpi, spi


@spi("mesh")
class Registry(ABC):

    @abstractmethod
    @mpi("mesh.registry.put")
    def register(self, registration: Registration):
        """
        Register metadata to mesh graph database.
        :param registration:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.registry.puts")
    def registers(self, registrations: List[Registration]):
        """
        Register metadata to mesh graph database.
        :param registrations:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.registry.remove")
    def unregister(self, registration: Registration):
        """
        Unregister metadata from mesh graph database.
        :param registration:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.registry.export")
    def export(self, kind: str) -> List[Registration]:
        """
        Export register metadata of mesh graph database.
        :param kind:
        :return:
        """
        pass

File: client/java/src/main/java/io/be/mesh/prsim/Registry.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.prsim;

import io.be.mesh.macro.Index;
import io.be.mesh.macro.MPI;
import io.be.mesh.macro.SPI;
import io.be.mesh.struct.Registration;

import java.util.List;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public interface Registry {

    @MPI("mesh.registry.put")
    void register(@Index(0) Registration registration);

    @MPI("mesh.registry.puts")
    void registers(@Index(0) List<Registration> registrations);

    @MPI("mesh.registry.remove")
    void unregister(@Index(0) Registration registration);

    @MPI("mesh.registry.export")
    List<Registration> export(@Index(0) String kind);
}
